pkg/core/auth/session/provider_impl_test.go

```diff
@@ -0,0 +1,152 @@
+				})
+
+				session2, err := provider.Create("user-id", "principal-id")
+				So(err, ShouldBeNil)
+				So(session2, ShouldResemble, &Session{
+					ID:                   session2.ID,
+					UserID:               "user-id",
+					PrincipalID:          "principal-id",
+					CreatedAt:            initialTime,
+					AccessedAt:           initialTime,
+					AccessToken:          session2.AccessToken,
+					AccessTokenCreatedAt: initialTime,
+				})
+
+				So(session1.ID, ShouldNotEqual, session2.ID)
+			})
+		})
+
+		Convey("getting session", func() {
+			fixtureSession := Session{
+				ID:                   "session-id",
+				UserID:               "user-id",
+				PrincipalID:          "principal-id",
+				CreatedAt:            initialTime,
+				AccessedAt:           initialTime,
+				AccessToken:          "session-id.access-token",
+				AccessTokenCreatedAt: initialTime,
+			}
+			store.Sessions["session-id"] = fixtureSession
+
+			Convey("should be successful", func() {
+				session, err := provider.GetByToken("session-id.access-token", TokenKindAccessToken)
+				So(err, ShouldBeNil)
+				So(session, ShouldResemble, &fixtureSession)
+			})
+
+			Convey("should reject non-existant session", func() {
+				session, err := provider.GetByToken("session-id-unknown.access-token", TokenKindAccessToken)
+				So(err, ShouldBeError, ErrSessionNotFound)
+				So(session, ShouldBeNil)
+			})
+
+			Convey("should reject incorrect token", func() {
+				session, err := provider.GetByToken("session-id.incorrect-token", TokenKindAccessToken)
+				So(err, ShouldBeError, ErrSessionNotFound)
+				So(session, ShouldBeNil)
+
+				session, err = provider.GetByToken("invalid-token", TokenKindAccessToken)
+				So(err, ShouldBeError, ErrSessionNotFound)
+				So(session, ShouldBeNil)
+			})
+		})
+
+		Convey("accessing session", func() {
+			session := Session{
+				ID:                   "session-id",
+				UserID:               "user-id",
+				PrincipalID:          "principal-id",
+				CreatedAt:            initialTime,
+				AccessedAt:           initialTime,
+				AccessToken:          "access-token",
+				AccessTokenCreatedAt: initialTime,
+			}
+			timeProvider.AdvanceSeconds(100)
+			timeNow := timeProvider.TimeNowUTC
+			store.Sessions["session-id"] = session
+
+			Convey("should be update accessed at time", func() {
+				err := provider.Access(&session)
+				So(err, ShouldBeNil)
+				So(session.AccessedAt, ShouldEqual, timeNow)
+			})
+		})
+
+		Convey("invalidating session", func() {
+			store.Sessions["session-id"] = Session{
+				ID:                   "session-id",
+				UserID:               "user-id",
+				PrincipalID:          "principal-id",
+				CreatedAt:            initialTime,
+				AccessedAt:           initialTime,
+				AccessToken:          "access-token",
+				AccessTokenCreatedAt: initialTime,
+			}
+
+			Convey("should be successful", func() {
+				err := provider.Invalidate("session-id")
+				So(err, ShouldBeNil)
+				So(store.Sessions, ShouldBeEmpty)
```

Comment:
Get once before invalidation and assert the session is there.
Get once after invalidation and assert the session is gone.

Reply:
I don't think it is needed? The parent block setup one session in the fake store, and each block assert the expected number of sessions in fake store after the operation.